src/transform/src/fusion/union.rs

```diff
@@ -76,7 +76,10 @@ impl Union {
                         _ => new_inputs.push(outer_input),
                     }
                 }
-                *relation = MirRelationExpr::union_many(new_inputs, relation_type);
+                // A valid relation type is only needed for empty unions, but the `any` call
+                // above guarantees the new union is never empty.
+                assert!(!new_inputs.is_empty());
```

Comment:
To be honest, I'm not sure I understand if the `any` call is even necessary; we are guaranteed to be in a `Union` which is guaranteed to have at least a `base`. Maybe not a problem for this diff, but just me trying to understand what the code needs to guarantee and why.

Reply:
Oh, good point. Yeah, the `any` call is not needed but handy. I'll update the comment saying that the non-emptiness of the union is guaranteed since it comes from an existing union, which cannot be empty.